### PR TITLE
kad.FindNear fix

### PR DESCRIPTION
**Pull request by wthorp**
We realized that the Kademlia FindNear() function was
1) not using XOR distance (AKA _totally broken_)
2) largely a duplicate of the RoutingTable FindNear() function

Changes in this PR:
1) upgraded RoutingTable FindNear() to use iterator and restrictions
2) removed unneeded RoutingTable interface
3) made Kademlia wrap methods that were previously accessed via RoutingTable
4) fixed the tests

### CODE REVIEW

File: pkg/kademlia/routing.go [hunk 1]
```diff
@@ -254,8 +259,26 @@ func (rt *RoutingTable) GetBucketTimestamp(bIDBytes []byte) (time.Time, error) {
...
+	return rt.nodeBucketDB.Iterate(storage.IterateOptions{First: storage.Key(start.Bytes()), Recurse: true},
+		func(it storage.Iterator) error {
+			var item storage.ListItem
+			for it.Next(&item) {
+				nodeID, err := storj.NodeIDFromBytes(item.Key)
+				if err != nil {
+					return err
+				}
+				if nodeID == rt.self.Id {
+					continue //todo:  revisit if self ID should be in routing table
```
Comment (jenlij) [inline pkg/kademlia/routing.go]: i think we may want to remove self from the results (see https://github.com/storj/storj/pull/1270/files#diff-06f20e1916ed6b711730b136a3f94975R163 )

File: pkg/dht/dht.go [hunk 2]
```diff
@@ -5,38 +5,20 @@ package dht
...
 	"storj.io/storj/pkg/pb"
 	"storj.io/storj/pkg/storj"
-	"storj.io/storj/storage"
 )
 
 // DHT is the interface for the DHT in the Storj network
 type DHT interface {
 	FindNear(ctx context.Context, start storj.NodeID, limit int, restrictions ...pb.Restriction) ([]*pb.Node, error)
-	GetRoutingTable() RoutingTable
 	Bootstrap(ctx context.Context) error
```
[the remark this answers is not in the document]
Comment (wthorp) [reply pkg/dht/dht.go]: Egon has been trying to get rid of these single-usage interfaces for a while, so... no?

File: pkg/kademlia/routing.go (hunk 1 above)
Comment (wthorp) [reply pkg/kademlia/routing.go]: Its good to see other evidence corroborating this!

File: pkg/dht/dht.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (jenlij) [reply pkg/dht/dht.go]: ah i guess since there's no other package using those methods then that's fine.

**Pull request merged**